### PR TITLE
Adding policies for Databricks workloads

### PR DESCRIPTION
**Pull request by krnese**
## Overview/Summary

Replace this with a brief description of what this Pull Request fixes, changes, etc.

## This PR fixes/adds/changes/removes

Adds the first iteration of data specific policies to landing zones, accelerating Enterprise-Scale Analytics

### Breaking Changes

N/A

## Testing Evidence

Please provide any testing evidence to show that your Pull Request works/fixes as described and planned (include screenshots, if appropriate).

## As part of this Pull Request I have

- [x] Checked for duplicate [Pull Requests](https://github.com/Azure/Enterprise-Scale/pulls)
- [x] Associated it with relevant [issues](https://github.com/Azure/Enterprise-Scale/issues), for tracking and closure.
- [x] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/Enterprise-Scale/tree/main)
- [x] Performed testing and provided evidence.
- [x] Updated relevant and associated documentation.
- [ ] Updated the ["What's New?"](https://github.com/Azure/Enterprise-Scale/wiki/Whats-new) wiki page (located: `/docs/wiki/whats-new.md`)

### CODE REVIEW

**Review by marvinbuss — DISMISSED** [review]
LGTM!

**Comment by jtracey93** [comment]
Can we please update the "What's New?" Wiki page please @krnese @marvinbuss 👍

**Comment by marvinbuss** [comment]
@jtracey93 Is this sufficient?

* Added Policies for [Azure Databricks](https://github.com/Azure/Enterprise-Scale/pull/757) or should we add a table with policy name and description?

File: eslzArm/eslz-portal.json [hunk 1]
```diff
@@ -2333,6 +2393,9 @@
                 "enableStorageHttps": "[steps('lzGoalState').esHttpsStorage]",
                 "denyIpForwarding": "[steps('lzGoalState').esIpForwarding]",
                 "denySubnetWithoutNsg": "[steps('lzGoalState').esNsg]",
+                "denyDatabricsPip": "[steps('lzGoalState').esLzDbPip]",
```
Comment (marvinbuss) [inline eslzArm/eslz-portal.json:2396]:
```suggestion
                "denyDatabricksPip": "[steps('lzGoalState').esLzDbPip]",
```

File: eslzArm/managementGroupTemplates/policyAssignments/Deny-DatabricksPublicIpPolicyAssignment.json [hunk 2]
```diff
@@ -0,0 +1,49 @@
+{
```
Comment (marvinbuss) [inline eslzArm/managementGroupTemplates/policyAssignments/Deny-DatabricksPublicIpPolicyAssignment.json:1]: The name is not consistent with the others `DENY`.

File: eslzArm/eslzArm.json [hunk 3]
```diff
@@ -598,6 +598,7 @@
             "managementGroups": "[uri(deployment().properties.templateLink.uri, 'managementGroupTemplates/mgmtGroupStructure/mgmtGroups.json')]",
```
Comment (marvinbuss) [inline eslzArm/eslzArm.json:598]: In this file we are still missing the input parameters as well as the deployment of the assignment, correct?

**Review by marvinbuss — CHANGES_REQUESTED** [review]
Please see my comments.

File: eslzArm/eslzArm.json (hunk 3 above)
Comment (krnese) [reply eslzArm/eslzArm.json:598]: This is the deployment of the dataPolicies.json (definitions) and the parameter for eslz prefix is provided. The assignment are separate deployments.

**Review by marvinbuss — APPROVED** [review]
LGTM!

**Comment by krnese** [comment]
@marvinbuss - can you pls make PR to update what's new?

**Pull request merged**